runtime/lib/cloudapi/index.js

```diff
@@ -72,18 +74,23 @@ CloudStore.prototype.connect = function connect (masterId) {
         if (!res.success) {
           throw new Error(`cloud login failed ${res.msg}(${res.code})`)
         }
-        handleResponse(JSON.parse(res.data))
-        return resolve(true)
+        return resolve(JSON.parse(res.data))
       } catch (err) {
         logger.error(err)
         return reject(err)
       }
     }
-  }).then(() => {
+  }).then((data) => {
     this.options.notify('101', STRINGS.LOGIN_DONE)
     this.options.notify('201', STRINGS.BIND_MASTER_DONE)
+    return this.handleResponse(data)
```

Comment:
Notify has been fired before response handling. Would this cause any side effects? /cc @yorkie

Reply:
Yea, it is, and notify is designed to get some events on the login progress.

Reply:
```js
// CloudStore.prototype.handleResponse
if (!basicInfo.master) {
  throw new Error('bind master is required')
}
```

**important**: There is a login process changed here. The function `handleResponse` should be executed before the `notify` function. Because the function `handleResponse` will check the response.